### PR TITLE
Simplify test workflow

### PR DESCRIPTION
**Pull request by ddundo**
Related to https://github.com/mesh-adaptation/docs/pull/90 (changes described there).

### CODE REVIEW

File: .github/workflows/test_suite.yml [hunk 1]
```diff
@@ -5,26 +5,24 @@ on:
   push:
     branches:
       - main
+    paths:
+      - '**.py'
+      - '**.msh'
+      - '**.geo'
+      - '.github/workflows/test_suite.yml'
```
Comment (jwallwork23) [inline .github/workflows/test_suite.yml:12]: Yeah it's probably fine not to test after such merges, given that we test every week anyway.

File: .github/workflows/test_suite.yml [hunk 2]
```diff
@@ -5,26 +5,24 @@ on:
...
-        python $(which firedrake-clean)
-        export GITHUB_ACTIONS_TEST_RUN=1
-        python -m coverage erase
-        python -m coverage run --source=goalie -m pytest -v --durations=20 test
-        python -m coverage report
-      changed-files-patterns: |
-        **/*.py
-        **/*.msh
-        **/*.geo
+    uses: mesh-adaptation/docs/.github/workflows/reusable_test_suite.yml@simplify_test
```
Comment (jwallwork23) [inline .github/workflows/test_suite.yml:28]: Reminder to update this to `main`.

**Review by jwallwork23 — APPROVED** [review]

File: .github/workflows/test_suite.yml (hunk 1 above)
Comment (ddundo) [reply .github/workflows/test_suite.yml:12]: Sorry, not sure if I understood. You'd like to remove these triggers from `push: main`?

Comment (jwallwork23) [reply .github/workflows/test_suite.yml:12]: Apologies - I wasn't that clear. My comment was saying what you've introduced is a good idea.

**Pull request merged**